Assets/Scripts/Turret/Shooting.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Cosmobot
+{
+    public class Shooting : MonoBehaviour
+    {
+        ParticleSystem ps;
+        Ray ray;
+        RaycastHit hit;
+        Tracking tracking;
+        TurretStats turretStats;
+        //public float fireRate = 2f;
+        //public float power = 12.5f;
+        bool canFire = true;
+        void Start()
+        {
+            tracking = GetComponent<Tracking>();
+            ps = GetComponent<ParticleSystem>();
+            turretStats = GetComponent<TurretStats>();
+        }
+
+        // Update is called once per frame
+        void Update()
+        {
+            ray = new Ray(transform.position, transform.forward);
+            if(Physics.Raycast(ray, out hit))
```

Comment:
`Physics.Raycast` is an expensive operation and should not be called inside `Update`. Try to invert logic: first `canFire` check and then raycast. There is chance that turret will be blocked by something, so it will keep casting raycasts every frame, but i think that would be too much optimization for now